giskard/scanner/llm/data_generation.py

```diff
@@ -0,0 +1,66 @@
+import json
+from typing import Optional
+
+import pandas as pd
+
+from giskard.scanner.llm import utils
+
+from ...datasets import Dataset
+from ...models.base.model import BaseModel
+from ..logger import logger
+
+GENERATE_INPUTS_PROMPT = """You are auditing an AI model. Your task is to generate typical but varied inputs for this model.
+
+Your will generate inputs for the following model: {model_name} - {model_description}.
+Model Features: {feature_names}
+
+Each generated input must be an object with values for each of the model features, aimed at simulating typical usage of the model, or typical inputs that the model is supposed to handle.
+Take the model description into account when generating the inputs. You should not generate repeated inputs or variations of the same input, instead try to generate inputs that varied for use cases of the model and cover all situations that could be encoutered during typical usage of the model.
+
+Think step by step and then call the `generate_inputs` function with the generated inputs. You must generate {num_samples} inputs.
+"""
+
+
+def generate_test_dataset(model: BaseModel, num_samples: int = 10, prompt: Optional[str] = None, temperature=0.5):
+    """Generates a test dataset for the model."""
+    functions = [
+        {
+            "name": "generate_inputs",
+            "description": "generates inputs for model audit",
+            "parameters": {
+                "type": "object",
+                "properties": {
+                    "inputs": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {name: {"type": "string"} for name in model.meta.feature_names},
```

Comment:
and if `feature_names=None`? 

I had to deal with the question as well, how about filling the `feature_names` based on:
```py
model.meta.feature_names = model.meta.feature_names or list(dataset.df.columns.drop(dataset.target, errors="ignore"))
```
during the validation phase? Otherwise we'd need the Dataset just for this.

Reply:
Well if `feature_names` is `None` we will have to get df columns. If none is provided we will not be able to scan the model, we should probably throw an error at the start of the scan saying we miss information to perform a scan

Reply:
yes but that's not what's done in `"properties": {name: {"type": "string"} for name in model.meta.feature_names}` is what I mean. This line would break in this case.

Reply:
There are few things we need to validate:

- Model must have a name
- Model must have a description
- Model must specify features

We need to add this extra validation at the beginning of the scan.